File: src/Jokenpo.java
import java.util.Map;

public class Jokenpo {

    private Algoritmo algoritmo;

    public void setAlgoritmo(Algoritmo algoritmo) {
        this.algoritmo = algoritmo;
    }

    public void jogar(Tipo ptipo) {
        Map<String, String> map = algoritmo.executar(ptipo);
        System.out.println(map);
    }
}

File: Jokenpo.py
from typing import Dict

from Algoritmo import Algoritmo
from TIpo import Tipo


class Jokenpo:
    def __init__(self):
        self.algoritmo = None

    def setAlgoritmo(self, algoritmo: Algoritmo):
        self.algoritmo = algoritmo

    def jogar(self, ptipo: Tipo):
        map_resultado: Dict[str, str] = self.algoritmo.executar(ptipo)
        print(map_resultado)
